utoipauto-core/src/discover.rs

```diff
@@ -86,7 +86,8 @@ fn parse_module_items(
                 .map(|item| DiscoverType::Fn(build_path(module_path, &item)))
                 .collect(),
             syn::Item::Struct(s) => {
-                let is_generic = !s.generics.params.is_empty();
+                let is_generic = !s.generics.params.iter().all(|p| matches!(p, syn::GenericParam::Lifetime(_)));
```

Comment:
Couldn't we match for all of GenericParam not only Lifetime?

I think this would fit better because I am not sure if only matching for lifetime will work.

Reply:
Nvm

I am stupid
Just realized my mistake. 🤦‍♂️

But I still think that matching only for lifetimes is not what we want